WordPress/gutenberg
# `UnitControl`: convert Storybook example to TypeScript and Controls

## What?


This PR converts `UnitControl`'s Storybook examples to TypeScript. While doing so, it also refactors those examples from using knobs to using controls.

## Why?


Part of the `@wordpress/components`'s TypeScript migration (#35744).

Altogether, the migration to TypeScript and the refactor to controls allow Storybook to extract prop types correctly and display them in the "docs" tab.

## How?


- Converted `UnitControl`'s story to TypeScript
- Added a named export for the un-forwarded `UnitControl` component (necessary to extract TS props correctly)
- Added a few missing types (`hideLabelFromVision`, `isShiftStepEnabled`, `shiftStep`) (these types should be later removed once `NumberControl` is migrated to TypeScript, since at that point `UnitControl` will be able to directly extend `NumberControl`'s props)
- Refactored from knobs to controls

## Testing Instructions





1. (optional) `npm run distclean && npm ci`
2. `npm run storybook:dev`
3. Compare [local `UnitControl` stories](http://localhost:50240/?path=/docs/components-experimental-unitcontrol--default) to the ones [currently on the stable branch](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-unitcontrol--default)
4. Make sure that the new docs are correct

## Screenshots or screencast 


https://user-images.githubusercontent.com/1083581/157500880-bed5c5a4-32fb-41ae-a03d-5d8e2ebb611d.mp4
